sparse_dot_topn/sparse_dot_topn_parallel.cpp

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Author: Zhe Sun, Ahmet Erdem
+// April 20, 2017
+
+#include <vector>
+#include <limits>
+#include <algorithm>
+#include <thread>
+
+unsigned int nthreads = std::thread::hardware_concurrency();
+
+#include "./sparse_dot_topn_parallel.h"
+
+struct candidate {int index; double value;};
+//
+static bool candidate_cmp(candidate c_i, candidate c_j) { return (c_i.value > c_j.value); }
+
+void inner_sparse_function(int start_row, int end_row, int n_col_inner,
```

Comment:
This seems like code duplication from the original function. Can you call the original function instead of copying the functionality?